### PR TITLE
Feat: Added ratings and snackbar

### PR DESCRIPTION
**Pull request by LordUttam**
Added Ratings and snackbar. Also made a fix in sidebar and separated JS for components. Please review, Thanks. :)
image
image
image

### CODE REVIEW

File: styles/button.css [hunk 1]
```diff
@@ -1,52 +1,51 @@
...
-	margin : 1rem;
-	border-radius : 5px;
-	font-size: 1.125rem;
-	letter-spacing: 1px;
-	cursor: pointer;
-	transition: all .2s;
+  padding: 1rem 2rem;
+  margin: 1rem;
+  border-radius: 5px;
+  font-size: 1.125rem;
```
Comment (Mohasin-Haque) [inline styles/button.css:3]: why not only 1.2 rem?

**Review by Mohasin-Haque — COMMENTED** [review]
Code looks good to me. Given some comments

**Review by yashpatel1797 — APPROVED** [review]
Looks good to merge.

File: styles/button.css (hunk 1 above)
Comment (LordUttam) [reply styles/button.css:3]: ok corrected

**Pull request merged**